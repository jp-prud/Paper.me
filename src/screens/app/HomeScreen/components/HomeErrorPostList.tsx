import {Box, Button, Text} from '@components';

interface HomeErrorPostListProps {
  refetch: () => void;
}

export function HomeErrorPostList({refetch}: HomeErrorPostListProps) {
  return (
    <Box flex={1} justifyContent="center" testID="home-error-post-list">
      <Box alignItems="center" gap="s4" mb="s24">
        <Text preset="paragraphLarge" bold>
          Oops! ðŸ™ˆ
        </Text>

        <Text textAlign="center">
          NÃ£o foi possÃ­vel carregar os posts. Por favor, verifique sua conexÃ£o e
          tente novamente.
        </Text>
      </Box>

      <Button text="Tentar novamente" onPress={refetch} />
    </Box>
  );
}
