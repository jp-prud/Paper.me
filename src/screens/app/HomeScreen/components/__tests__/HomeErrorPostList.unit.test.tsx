import {render, fireEvent} from '@tests';

import {HomeErrorPostList} from '../HomeErrorPostList';

describe('HomeErrorPostList - unit', () => {
  const refetchMock = jest.fn();

  it('should render the component', () => {
    const {getByTestId, toJSON} = render(
      <HomeErrorPostList refetch={refetchMock} />,
    );

    expect(getByTestId('home-error-post-list')).toBeDefined();
    expect(toJSON()).toMatchSnapshot();
  });

  it('should call refetch when click on CTA', () => {
    const {getByText} = render(<HomeErrorPostList refetch={refetchMock} />);

    const button = getByText('Tentar novamente');

    fireEvent.press(button);

    expect(refetchMock).toHaveBeenCalledTimes(1);
  });
});
