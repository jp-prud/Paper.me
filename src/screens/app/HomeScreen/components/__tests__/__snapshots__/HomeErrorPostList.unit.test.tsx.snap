// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeErrorPostList - unit should render the component 1`] = `
[
  <View
    style={
      [
        {
          "flex": 1,
          "justifyContent": "center",
        },
      ]
    }
    testID="home-error-post-list"
  >
    <View
      style={
        [
          {
            "alignItems": "center",
            "gap": 4,
            "marginBottom": 24,
          },
        ]
      }
    >
      <Text
        selectionColor="#c7e7ea"
        style={
          [
            {
              "color": "#262626",
            },
            [
              {
                "fontSize": 18,
                "lineHeight": 25.2,
              },
              {
                "fontFamily": "PlusJakartaSans-Bold",
              },
              undefined,
            ],
          ]
        }
      >
        Oops! ðŸ™ˆ
      </Text>
      <Text
        selectionColor="#c7e7ea"
        style={
          [
            {
              "color": "#262626",
              "textAlign": "center",
            },
            [
              {
                "fontSize": 16,
                "lineHeight": 22.4,
              },
              {
                "fontFamily": "PlusJakartaSans-Regular",
              },
              undefined,
            ],
          ]
        }
      >
        NÃ£o foi possÃ­vel carregar os posts. Por favor, verifique sua conexÃ£o e tente novamente.
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#017582",
          "borderRadius": 16,
          "height": 50,
          "justifyContent": "center",
          "opacity": 1,
          "paddingHorizontal": 20,
        }
      }
    >
      <Text
        selectionColor="#c7e7ea"
        style={
          [
            {
              "color": "#FFFFFF",
            },
            [
              {
                "fontSize": 16,
                "lineHeight": 22.4,
              },
              {
                "fontFamily": "PlusJakartaSans-Bold",
              },
              undefined,
            ],
          ]
        }
      >
        Tentar novamente
      </Text>
    </View>
  </View>,
  " ",
]
`;
